src/main/java/edu/cornell/kfs/sys/dataaccess/impl/HistoricalTablePurgeRecordsDaoOjb.java

```diff
@@ -0,0 +1,144 @@
+import org.apache.logging.log4j.Logger;
+import org.apache.ojb.broker.query.Criteria;
+import org.apache.ojb.broker.query.QueryFactory;
+import org.joda.time.DateTime;
+import org.kuali.kfs.krad.bo.PersistableBusinessObjectBase;
+import org.kuali.kfs.krad.service.BusinessObjectService;
+import org.kuali.kfs.sys.util.KfsDateUtils;
+import org.kuali.kfs.core.framework.persistence.ojb.dao.PlatformAwareDaoBaseOjb;
+import org.kuali.kfs.coreservice.framework.parameter.ParameterService;
+import org.springframework.transaction.annotation.Transactional;
+
+import edu.cornell.kfs.sys.CUKFSParameterKeyConstants;
+import edu.cornell.kfs.sys.batch.service.HistoricalTableLookupCriteriaPurgeService;
+import edu.cornell.kfs.sys.businessobject.HistoricalTableDetailsForPurge;
+import edu.cornell.kfs.sys.businessobject.PurgeableBusinessObjectInterface;
+import edu.cornell.kfs.sys.dataaccess.HistoricalTablePurgeRecordIdentifiersDao;
+import edu.cornell.kfs.sys.dataaccess.HistoricalTablePurgeRecordsDao;
+
+/**
+ * CU Customization: Historical Purge Table Job : short lived batch job to clear backlog of Initiated documents.
+ *
+ */
+
+public class HistoricalTablePurgeRecordsDaoOjb extends PlatformAwareDaoBaseOjb implements HistoricalTablePurgeRecordsDao {
+    private static final Logger LOG = LogManager.getLogger(); 
+    private BusinessObjectService businessObjectService;
+    protected ParameterService parameterService;
+    private HistoricalTablePurgeRecordIdentifiersDao historicalTablePurgeRecordIdentifiersDao;
+    
+    @Override
+    public void purgeRecords(java.util.Date jobRunDate, List<HistoricalTableDetailsForPurge> tableDetails) {
+        int defaultDaysOld = retrieveDefaultDaysBeforePurgeParameterValue();
+        
+        for (HistoricalTableDetailsForPurge details : tableDetails) {
+            int daysOldToUse = details.isUseDefaultDaysBeforePurgeParameter() ? defaultDaysOld : retrieveDaysBeforePurgeParameterValue(details);
+            int databaseRowFetchCount = retrieveDatabaseRowFetchCountParameterValue(details);
+            Date dateForPurge = getPurgeDate(jobRunDate, daysOldToUse);
+            LOG.info("purgeRecords: daysOldToUse = {}    databaseRowFetchCount = {}    dateForPurge = {}", daysOldToUse, databaseRowFetchCount, dateForPurge);
+            
+            List<String> initiatedDocIdsToPurge = historicalTablePurgeRecordIdentifiersDao.obtainInitiatedDocumentIdsToPurge(dateForPurge, databaseRowFetchCount);
+            
+            for (String documentIdToPurge : initiatedDocIdsToPurge) {
+                Criteria lookupCriteria = buildTablePurgeCriteria(details.getServiceImplForPurgeTableLookupCriteria(), documentIdToPurge);
+                identifyAndRequestRecordsDeletion(details, lookupCriteria);
+            }
+        }
+    }
+
+    protected Criteria buildTablePurgeCriteria(HistoricalTableLookupCriteriaPurgeService serviceImplClassForPurgeTableLookupCriteria, String documentIdToPurge) {
```

Comment:
Would it be better just to call serviceImplClassForPurgeTableLookupCriteria.buildLookupCriteria() rather than wrap it in this function?  it doesn't seem to add much clarity.